python/lsst/ip/diffim/getTemplate.py

```diff
@@ -107,19 +113,32 @@ def run(self, exposure, sensorRef, templateIdList=None):
                 bbox=patchSubBBox,
                 tract=tractInfo.getId(),
                 patch="%s,%s" % (patchInfo.getIndex()[0], patchInfo.getIndex()[1]),
+                numSubfilters=self.config.numSubfilters,
             )
             if patchSubBBox.isEmpty():
                 self.log.info("skip tract=%(tract)s, patch=%(patch)s; no overlapping pixels" % patchArgDict)
                 continue
-            if not sensorRef.datasetExists(**patchArgDict):
-                self.log.warn("%(datasetType)s, tract=%(tract)s, patch=%(patch)s does not exist"
-                              % patchArgDict)
-                continue
 
             nPatchesFound += 1
-            self.log.info("Reading patch %s" % patchArgDict)
-            coaddPatch = sensorRef.get(**patchArgDict)
-            coaddExposure.getMaskedImage().assign(coaddPatch.getMaskedImage(), coaddPatch.getBBox())
+            if self.config.coaddName == 'dcr':
```

Comment:
I not crazy about the "if dcr do these three things else do these other three things."   It's fine for now, but will you help me refactor it per RFC-352 soon?

Reply:
I quite agree that the "if dcr..." block is ugly, and I would strongly welcome any suggestions for how to change it. One possibility would be to modify the butler so that `coaddPatch = sensorRef.get(**patchArgDict)` would know to calculate and return the same thing as the DcrModel calculations, but I worry that would look surprising and magic.

Reply:
I am also happy to help implement RFC-352 if that is useful, though `DcrModel` here is not a `Task`, so I'm not sure what needs to be done.

Reply:
I meant that I plan to refactor `GetCoaddAsTemplateTask`.  I might need to create a DcrModel with your `fromImage` interface rather than your `fromDataRef` interface, and might need advice.